pkg/metrics/check_metrics.go

```diff
@@ -0,0 +1,158 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/ckey"
+	"github.com/DataDog/datadog-agent/pkg/telemetry"
+)
+
+var (
+	tlmCheckMetricsTotal   = telemetry.NewGauge("check_metrics", "total_metrics", []string{"stateful"}, "Gauge of all check metrics")
+	tlmCheckMetricsActive  = telemetry.NewGauge("check_metrics", "active_metrics", []string{"stateful"}, "Gauge of non-expired check metrics (total - waiting)")
+	tlmCheckMetricsWaiting = telemetry.NewGauge("check_metrics", "waiting_metrics", []string{"stateful"}, "Gauge of expired check metrics waiting for timeout")
+	tlmCheckMetricsAdded   = telemetry.NewCounter("check_metrics", "created_total", []string{"stateful"}, "Count of new check metrics added")
+	tlmCheckMetricsExpired = telemetry.NewCounter("check_metrics", "expired_total", []string{"stateful"}, "Count of expired metrics")
+	tlmCheckMetricsRemoved = telemetry.NewCounter("check_metrics", "removed_total", []string{"stateful"}, "Count of removed metrics")
+
+	checkMetricsAddSampleTelemetry = &AddSampleTelemetry{
+		Total:     tlmCheckMetricsAdded.WithValues("sum"),
+		Stateful:  tlmCheckMetricsAdded.WithValues("true"),
+		Stateless: tlmCheckMetricsAdded.WithValues("false"),
+	}
+	tlmCheckMetricsExpiredTotal     = tlmCheckMetricsExpired.WithValues("sum")
```

Comment:
Why record this "sum" tlm in addition to its parts?  Couldn't the app perform that summation?